sdk/core/src/policies/transport.rs

```diff
@@ -37,6 +50,184 @@ where
+        const SKIPPED_HEADERS: &[&'static str] = &["Date", "x-ms-date", "authorization"];
+        let received_headers_to_match = request
+            .headers()
+            .iter()
+            .filter(|h| {
+                SKIPPED_HEADERS
+                    .iter()
+                    .find(|to_skip| to_skip == &&h.0.as_str())
+                    .is_none()
+            })
+            .collect::<Vec<_>>();
+
+        let read_headers_to_match = read_request
+            .headers()
+            .iter()
+            .filter(|h| {
+                SKIPPED_HEADERS
+                    .iter()
+                    .find(|to_skip| to_skip == &&h.0.as_str())
+                    .is_none()
+            })
+            .collect::<Vec<_>>();
+
+        // In order to accept a request, we make sure that:
+        // 1. There are no extra headers (in both the received and read request).
+        // 2. Each header has the same value.
+        if received_headers_to_match.len() != read_headers_to_match.len() {
+            panic!(
+                "different number of headers in request. Recevied: {}, Read: {}",
+                received_headers_to_match.len(),
+                read_headers_to_match.len()
+            );
+        }
+
+        for received_header_to_match in received_headers_to_match.iter() {
+            let read_header_to_match = read_headers_to_match
+                .iter()
+                .find(|h| received_header_to_match.0.as_str() == h.0.as_str())
+                .expect(&format!(
+                    "received request have header {} but it was not present in the read request",
+                    received_header_to_match.0.as_str()
+                ));
+
+            if received_header_to_match.1 != read_header_to_match.1 {
+                panic!(
+                    "header {} value is different. Received: {}, Read: {}",
+                    received_header_to_match.0.as_str(),
+                    received_header_to_match.1.to_str().unwrap(),
+                    read_header_to_match.1.to_str().unwrap()
+                );
+            }
+        }
+
+        if read_request.method() != request.method() {
+            panic!(
+                "HTTP method is different. Received: {}, Read: {}",
+                read_request.method(),
+                request.method()
+            );
+        }
+
+        let received_body = match request.body() {
+            crate::Body::Bytes(bytes) => &bytes as &[u8],
+            crate::Body::SeekableStream(_) => unimplemented!(),
+        };
+
+        let read_body = match read_request.body() {
+            crate::Body::Bytes(bytes) => &bytes as &[u8],
+            crate::Body::SeekableStream(_) => unimplemented!(),
+        };
+
+        if received_body != read_body {
+            panic!(
+                "request body is different. Received: {}, Read: {}",
+                read_request.method(),
+                request.method()
```

Comment:
Typo here. We want to print the bodies and not the methods.

Reply:
Thanks that have been a PITA for the framework users to figure out!